### PR TITLE
JAMES-2949 Standardize case support for UsersRepository

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[8c857d845da8f993097b89534a9dd06b293e6e4a] [ERROR]   UserRoutesTest$WithVirtualHosting.putShouldFailOnRepositoryExceptionOnGetUserByName » Mockito
[8c857d845da8f993097b89534a9dd06b293e6e4a] [ERROR]   UserRoutesTest$WithoutVirtualHosting.putShouldFailOnRepositoryExceptionOnGetUserByName » Mockito
```

**Comment by chibenwa** [comment]
I HATE mocking!

File: server/data/data-cassandra/src/test/java/org/apache/james/user/cassandra/CassandraUsersRepositoryTest.java [hunk 1]
```diff
@@ -48,8 +49,11 @@ public void tearDown() throws Exception {
         cassandra.closeCluster();
     }
 
-    @Override
-    protected AbstractUsersRepository getUsersRepository() {
-        return new CassandraUsersRepository(domainList, cassandra.getConf());
+    @Override    protected AbstractUsersRepository getUsersRepository() throws Exception {
```
Comment (rouazana) [inline server/data/data-cassandra/src/test/java/org/apache/james/user/cassandra/CassandraUsersRepositoryTest.java]: missing cr/lf

**Comment by rouazana** [comment]
pushed forced for rebase on master

**Comment by Arsnael** [comment]
```
[ERROR]   QuotaUsageUpdatedEventSerializationTest.toJsonShouldReturnQuotaEventJson:81 JSON documents are different:
[31c288ebb6e42ba84fbbb637978b6a023b7894e4] Different value found in node "QuotaUsageUpdatedEvent.user", expected: <"onlyUsername"> but was: <"onlyusername">.
```
case issue?

**Comment by chibenwa** [comment]
```
[dcbfedd56dd7724070566ae059945f25995cc3fd] [ERROR]   TestScriptedUserAdder.testShouldExecuteScriptAgainstPort:52 
[dcbfedd56dd7724070566ae059945f25995cc3fd] Expecting:
[dcbfedd56dd7724070566ae059945f25995cc3fd]  <"USER='a user' password='Some Password'
[dcbfedd56dd7724070566ae059945f25995cc3fd] ">
[dcbfedd56dd7724070566ae059945f25995cc3fd] to be equal to:
[dcbfedd56dd7724070566ae059945f25995cc3fd]  <"USER='A User' password='Some Password'
[dcbfedd56dd7724070566ae059945f25995cc3fd] ">
[dcbfedd56dd7724070566ae059945f25995cc3fd] but was not.
```

File: server/data/data-jmap-cassandra/src/test/resources/json/event.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "type":"filtering-rule-set-defined",
   "eventId":0,
-  "aggregateId":"FilteringRule/Bart",
+  "aggregateId":"FilteringRule/bart",
```
Comment (chibenwa) [inline server/data/data-jmap-cassandra/src/test/resources/json/event.json]: Let say that both cassandra and jmap-draft are "experimental" and thus that users with capital letters loosing their "jmap filters" is oK ^^

Comment (remk) [reply server/data/data-jmap-cassandra/src/test/resources/json/event.json]: Should we document it somewhere ?

**Comment by rouazana** [comment]
```
[941bd9cfc2b94befd120768c705c9b0e7355ba34] [ERROR] Failed to execute goal on project james-server-lifecycle-api: Could not resolve dependencies for project org.apache.james:james-server-lifecycle-api:jar:3.5.0-SNAPSHOT: Failed to collect dependencies at com.google.guava:guava:jar:25.1-jre: Failed to read artifact descriptor for com.google.guava:guava:jar:25.1-jre: Could not transfer artifact com.google.guava:guava:pom:25.1-jre from/to central (https://repo.maven.apache.org/maven2): /root/.m2/repository/com/google/guava/guava/25.1-jre/guava-25.1-jre.pom.part (No such file or directory) -> [Help 1]
```
test this please

File: server/data/data-jmap-cassandra/src/test/resources/json/event.json (hunk 2 above)
Comment (chibenwa) [reply server/data/data-jmap-cassandra/src/test/resources/json/event.json]: We should document anyway that users with lower case will need MUA reconfiguration on top of JPA-Guice.

**Comment by rouazana** [comment]
test this please

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**